### PR TITLE
Test for bad return value

### PR DESCRIPTION
**Pull request by linas**
Add unit tests for GroundedSchema / GroundedPredicate's that return bad values.
 * Scheme code was already handling these properly.
 * Python was handling some of these properly, but not all.
   In some cases, it threw opaque, hard to understand errors.
   In the case of #2627 it was not noticing a bad return value.

Also, consolidate two different python execution paths into one.
They were cut-n-paste copies of each-other. Merge them back
together again. This results in an error being throw for #2627

### CODE REVIEW

File: opencog/cython/PythonEval.cc [hunk 1]
```diff
@@ -985,114 +985,76 @@ PyObject* PythonEval::call_user_function(const std::string& moduleFunction,
...
         throw RuntimeException(TRACE_INFO,
-            "Python function '%s' did not return TruthValue!",
+            "Python function '%s' did not return Atomese!",
             func.c_str());
 
     // Grab the GIL.
     PyGILState_STATE gstate = PyGILState_Ensure();
 
+    // Did we actually get a Value?
+    if (0 == PyObject_HasAttrString(pyValue, "value_ptr"))
```
Comment (vsbogd) [inline opencog/cython/PythonEval.cc:1015]: Checking if `pyValue` is subclass of the `Value` would be more safe. This is few more lines of code though.

Comment (vsbogd) [reply opencog/cython/PythonEval.cc:1015]: Other code looks good to me.

Comment (linas) [reply opencog/cython/PythonEval.cc:1015]: pyValue isn't a subclass of Value, its a pointer to a python object holding a pointer to a value.

Comment (linas) [reply opencog/cython/PythonEval.cc:1015]: Unless you mean that I should call `PyObject_IsInstance()` ...

Comment (vsbogd) [reply opencog/cython/PythonEval.cc:1019]: Yes, I mean `PyObject_IsInstance`

**Pull request merged**